charts/jenkins/templates/_helpers.tpl

```diff
@@ -480,3 +480,60 @@ Create the HTTP port for interacting with the controller
     {{- .Values.controller.targetPort -}}
 {{- end -}}
 {{- end -}}
+
+{{- define "jenkins.configReloadContainer" -}}
+{{- $root := index . 0 -}}
+{{- $containerName := index . 1 -}}
+{{- $method := index . 2 -}}
+- name: {{ $containerName }}
+  image: "{{ $root.Values.controller.sidecars.configAutoReload.image }}"
+  imagePullPolicy: {{ $root.Values.controller.sidecars.configAutoReload.imagePullPolicy }}
+  {{- if $root.Values.controller.sidecars.configAutoReload.containerSecurityContext }}
+  securityContext: {{- toYaml $root.Values.controller.sidecars.configAutoReload.containerSecurityContext | nindent 4 }}
+  {{- end }}
+  {{- if $root.Values.controller.sidecars.configAutoReload.envFrom }}
+  envFrom:
+{{ (tpl (toYaml $root.Values.controller.sidecars.configAutoReload.envFrom) $root) | indent 4 }}
+  {{- end }}
+  env:
+    - name: POD_NAME
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.name
+    - name: LABEL
+      value: "{{ template "jenkins.fullname" $root }}-jenkins-config"
+    - name: FOLDER
+      value: "{{ $root.Values.controller.sidecars.configAutoReload.folder }}"
+    - name: NAMESPACE
+      value: '{{ $root.Values.controller.sidecars.configAutoReload.searchNamespace | default (include "jenkins.namespace" $root) }}'
+{{- if $method }}
+    - name: METHOD
+      value: "{{ $method }}"
+{{- end }}
+{{- if $root.Values.controller.sidecars.configAutoReload.sleepTime }}
```

Comment:
Should we use an else if here to ensure that we can never end up with two `METHOD` values?

Reply:
Does https://github.com/jenkinsci/helm-charts/pull/926/commits/9d3a61e45d29fe5e7aa876e2cf8e086f392837f1 work?